### PR TITLE
fix: replace isDeprecated with isRemoved

### PR DESCRIPTION
**Pull request by klkucaj**
## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [x] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

### CODE REVIEW

**Review by SandPod — CHANGES_REQUESTED** [review]
Didn't finish the full review but I need you to take a closer look at what was actually deprecated and remove test related to this along with other functinality that is not needed.

File: tools/serverpod_cli/lib/src/analyzer/models/yaml_definitions/class_yaml_definition.dart [hunk 1]
```diff
@@ -57,12 +57,9 @@ class ClassYamlDefinition {
...
                 Keyword.parent,
                 isDeprecated: true,
-                mutuallyExclusiveKeys: {
-                  Keyword.relation,
-                },
+                isRemoved: true,
                 alternativeUsageMessage:
                     'Use the relation keyword instead. E.g. relation(parent=parent_table). Note that the default onDelete action changes from "Cascade" to "NoAction" when using the relation keyword.',
-                valueRestriction: restrictions.validateParentName,
               ),
```
[the remark this answers is not in the document]
Comment (klkucaj) [reply tools/serverpod_cli/lib/src/analyzer/models/yaml_definitions/class_yaml_definition.dart:63]: Its on line 74 now with the new update, should i remove it at all ?

[the remark this answers is not in the document]
Comment (SandPod) [reply tools/serverpod_cli/lib/src/analyzer/models/yaml_definitions/class_yaml_definition.dart:63]: No, if it is used in another validation rule it should not be removed

File: tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_parent_test.dart [hunk 2]
```diff
@@ -25,60 +24,6 @@ void main() {
           .validateAll();
 
       expect(collector.errors, isNotEmpty);
-
```
Comment (SandPod) [inline tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_parent_test.dart:28]: chore: This file still contains a bunch of tests that are not required anymore. Please remove all tests except maybe a single one that validates that the parent keyword generates an error deprecation warning.

You have basically removed the expects from all tests in the file that make them unique, so we are just validating the same thing over and over again.

File: tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_persist_test.dart [hunk 3]
```diff
@@ -388,18 +383,11 @@ void main() {
       StatefulAnalyzer(config, models, onErrorsCollector(collector))
           .validateAll();
 
-      expect(collector.errors.length, greaterThan(1));
-
-      var error1 = collector.errors[0];
-      var error2 = collector.errors[1];
+      expect(collector.errors.length, 1);
```
Comment (SandPod) [inline tools/serverpod_cli/test/analyzer/models/stateful_analyzer/model_validation/field/field_persist_test.dart:386]: chore: this test is reduntant remove it. Please check all other tests. We don't have any mutually exclusivity anymore so there is nothing to validate.

**Review by SandPod — CHANGES_REQUESTED** [review]
There are still reduntant tests connected to the removed keywords.

Please look at the complete diff of the PR and all tests you touched and ensure actually validate something.

**Review by SandPod — APPROVED** [review]
Looks good! Nice to get rid of the deprecations and good job navigating the test changes 🚀

**Pull request merged**